src/ccall.cpp

```diff
@@ -537,46 +537,6 @@ static void interpret_symbol_arg(jl_codectx_t &ctx, native_sym_arg_t &out, jl_va
     }
 }
 
-
-static void jl_rethrow_with_add(const char *fmt, ...)
-{
-    jl_value_t *exc = jl_current_exception();
-    if (jl_typeis(exc, jl_errorexception_type)) {
-        char *str = jl_string_data(jl_fieldref(exc, 0));
-        char buf[1024];
-        va_list args;
-        va_start(args, fmt);
-        int nc = vsnprintf(buf, sizeof(buf), fmt, args);
-        va_end(args);
-        nc += snprintf(buf + nc, sizeof(buf) - nc, ": %s", str);
-        jl_value_t *msg = jl_pchar_to_string(buf, nc);
-        JL_GC_PUSH1(&msg);
-        jl_throw(jl_new_struct(jl_errorexception_type, msg));
-    }
-    jl_rethrow();
-}
-
-static jl_value_t* try_eval(jl_codectx_t &ctx, jl_value_t *ex, const char *failure)
-{
-    jl_value_t *constant = static_eval(ctx, ex, true, true);
-    if (jl_is_ssavalue(ex) && !constant)
-        jl_error(failure);
-    else if (constant)
-        return constant;
-
-    JL_TRY {
-        size_t last_age = jl_get_ptls_states()->world_age;
-        jl_get_ptls_states()->world_age = ctx.world;
-        constant = jl_interpret_toplevel_expr_in(ctx.module, ex, ctx.source, ctx.linfo->sparam_vals);
```

Comment:
This functionality was unused, and seemed broken anyhow:

```
julia> f34166(x) = Base.llvmcall("ret i$(Sys.WORD_SIZE) %0", Int, (Int,), x)
f34166 (generic function with 1 method)

julia> f34166(1)
Internal error: encountered unexpected error during compilation of f34166:
ErrorException("error statically evaluating llvmcall argument tuple")
jl_error at /home/tim/Julia/julia/src/rtutils.c:41
try_eval at /home/tim/Julia/julia/src/ccall.cpp:563
emit_llvmcall at /home/tim/Julia/julia/src/ccall.cpp:679

signal (11): Segmentation fault
in expression starting at REPL[3]:1
eval_value at /home/tim/Julia/julia/src/interpreter.c:169
do_call at /home/tim/Julia/julia/src/interpreter.c:116
eval_value at /home/tim/Julia/julia/src/interpreter.c:206
jl_interpret_toplevel_expr_in at /home/tim/Julia/julia/src/interpreter.c:679
try_eval at /home/tim/Julia/julia/src/ccall.cpp:570
emit_llvmcall at /home/tim/Julia/julia/src/ccall.cpp:665
```

Is my understanding correct that the first attempt is during compilation of `f34166`, which throws, so we end up interpreting `f34166` which then causes a second compilation of the llvmcall?